src/neg_multiply.rs

```diff
@@ -0,0 +1,57 @@
+use rustc::hir::*;
+use rustc::lint::*;
+use syntax::codemap::{Span, Spanned};
+
+use consts::{self, Constant};
+use utils::span_lint;
+
+/// **What it does:** Checks for multiplication by -1 as a form of negation.
+///
+/// **Why is this bad?** It's more readable to just negate.
+///
+/// **Known problems:** This assumes that the type can be negated.
```

Comment:
Yeah, this text was from the workshop, when we didn't have the type test yet, and I forgot to change it.

We may want to extend to other known types where `x * -1 == -x` (i.e. `f32`, `f64`, perhaps some SIMD types?).